drivers/gpio/gpio_grgpio2.c

```diff
@@ -0,0 +1,304 @@
+		regs->imask_or = mask;
+	} else {
+		ret = -ENOTSUP;
+	}
+	k_spin_unlock(&data->lock, key);
+
+	/* Remove old interrupt history for this pin. */
+	regs->iflag = mask;
+
+	int interrupt = cfg->interrupt;
+	const int irqgen = data->irqgen;
+
+	if (irqgen == 0) {
+		interrupt += pin;
+	} else if (irqgen == 1) {
+		;
+	} else if (irqgen < 32) {
+		/* look up interrupt number in GRGPIO interrupt map */
+		uint32_t val = regs->irqmap[pin/4];
+
+		val >>= (3 - pin % 4) * 8;
+		interrupt += (val & 0x1f);
+	}
+
+	if (interrupt && ((1 << interrupt) & data->connected) == 0) {
+		irq_connect_dynamic(
+			interrupt,
+			0,
+			(void (*)(const void *)) grgpio_isr,
+			dev,
+			0
+		);
+		irq_enable(interrupt);
+		data->connected |= 1 << interrupt;
+	}
+
+	return ret;
+}
+
+static int manage_callback(const struct device *dev,
+			   struct gpio_callback *callback,
+			   bool set)
+{
+	struct data *data = dev->data;
+
+	return gpio_manage_callback(&data->cb, callback, set);
+}
+
+static void grgpio_isr(const struct device *dev)
+{
+	const struct cfg *cfg = dev->config;
+	struct data *data = dev->data;
+	volatile struct grgpio_regs *regs = cfg->regs;
+	uint32_t pins;
+
+	/* no locking needed when iflag is implemented */
+	pins = regs->iflag;
+	if (pins == 0) {
+		return;
+	}
+	regs->iflag = pins;
+	gpio_fire_callbacks(&data->cb, dev, pins);
+}
+
+static int grgpio_init(const struct device *dev)
+{
+	const struct cfg *cfg = dev->config;
+	struct data *data = dev->data;
+	volatile struct grgpio_regs *regs = cfg->regs;
+
+	data->irqgen = (regs->cap & GRGPIO_CAP_IRQGEN) >> GRGPIO_CAP_IRQGEN_BIT;
+	regs->dir = 0;
+	/* Mask all Interrupts */
+	regs->imask = 0;
+	/* Make IRQ Rising edge triggered default */
+	regs->ipol = 0xffffffff;
+	regs->iedge = 0xffffffff;
+	regs->iflag = 0xffffffff;
+	/* Read what I/O lines have IRQ support */
+	data->imask = regs->ipol;
+
+	return 0;
+}
+
+static const struct gpio_driver_api driver_api = {
+	.pin_configure                  = pin_configure,
+	.port_get_raw                   = port_get_raw,
+	.port_set_masked_raw            = port_set_masked_raw,
+	.port_set_bits_raw              = port_set_bits_raw,
+	.port_clear_bits_raw            = port_clear_bits_raw,
+	.port_toggle_bits               = port_toggle_bits,
+	.pin_interrupt_configure        = pin_interrupt_configure,
+	.manage_callback                = manage_callback,
+	.get_pending_int                = get_pending_int,
+};
+
+#define THE_INIT(n)							\
+	static const struct cfg cfg_##n = {				\
+		.common = {						\
+			.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(n),\
+		},							\
+		.regs = (void *) DT_INST_REG_ADDR(n),			\
+		.interrupt = DT_INST_IRQN(n),				\
+	};								\
+	static struct data data_##n;					\
+									\
+	DEVICE_DT_INST_DEFINE(n,					\
+			    grgpio_init,				\
+			    NULL,					\
+			    &data_##n,					\
+			    &cfg_##n,					\
+			    POST_KERNEL,				\
+			    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,		\
```

Comment:
```suggestion
			    CONFIG_GPIO_INIT_PRIORITY,		\
```

Reply:
Updated according to your recommendation.